### ISSUE TITLE
Promyvion - Vahzl BCNM: Procreator fission usage

### ISSUE THREAD

**Issue by DeuceFFXI** [EdenServer/community#5371]
### Checklist

- [x] I have searched for existing issues for issues like this one. The issue has not been posted.
- [x] I have provided reproducible steps. (No "as the title says" posts please. ***Please try to state facts only, and avoid assumptions/opinions.***)
- [x] I have provided my client version in the details. (type /ver into your game window)
- [x] ***I have provided proof of my claim (era videos, links to era wiki and/or forum posts)*** Lack of proof will slow down fixes and prioritization. Please provide details so that developers' research time is minimized.

### Details

ver 30181205_0

As the title suggests, went with a party to tackle the Promyvion - Vahzl BCNM fight earlier and we encountered a bug related to Procreator and its Fission TP move.  For the fight we elected to use two animas, Hystero and Pyscho, around 50% just prior to the first time the NM used Fission.  As can be seen in the two screenshots below, both animas had been used; however, the NM still used Fission within the 30 second timeframe:

![promyVahzl_bug_report1](https://user-images.githubusercontent.com/61279359/202123464-ba24831d-5964-4eae-85d7-2a17787901dc.png)
![promyVahzl_bug_report2](https://user-images.githubusercontent.com/61279359/202123472-163db119-f1cd-42d9-b65a-8b36b8f1cba1.png)

According to the classic wiki page for [Hysteroanima](https://classicffxi.fandom.com/wiki/Hysteroanima?so=search) the item is supposed to prevent the usage of TP moves, of which Fission is one.  It did not appear as though any other TP moves were used by Procreator while the effect of the Hysteroanima was active.  It seems that Fission either was still interrupted as normal, but the add still spawned, or the TP move completed when it shouldn't have.

**Comment by DeuceFFXI** [comment]
So, I did some digging on YouTube and have come to the determination that Fission in Promyvion needs an audit on Eden.  Currently on Eden, from my observations, Fission appears to:

1. Be used only at or below 50% of mob HP
2. Is used too frequently
3. Is used or rather queued while the mob is under the effect of a Hysteroanima

**Issue #1**, we'll look at two videos from YouTube, one from Promyvion - Dem and Promyvion - Vahzl, which indicate that the Fission TP move is accessible by the Procreator at any time, regardless of mobHP.
a. [Promyvion - Dem](https://youtu.be/LCtaeA4zwVM?t=196) In this video Procreator not only uses Fission before 50%, but also its first TP move!
b. [Promyvion - Vahzl](https://youtu.be/VMOhOT8eK8s?t=301) Here, Procreator uses Fission around 70%.

**Issue #2** is a little more difficult to narrow down; however, based on searching the logs from tackling Promyvion - Vahzl the other night and watching several videos from retail uploaded ~12-15 years ago, I get the impression Fission is used far too frequently on Eden.

![FissionLogs1](https://user-images.githubusercontent.com/61279359/202769480-9e73297a-521b-492c-9f20-7e45269aff28.png)
This image contains TP moves used by Procreator from two separate attempts.

Several videos of Promyvion - Dem & Vahzl from retail suggest Fission is not used as often as can be seen in these logs.  In both attempts on Eden, it seems as though Procreator effectively spammed the Fission TP move below 50%.

Promyvion - Dem
https://www.youtube.com/watch?v=VGVeyDysCgk - Disturbed song for audio, Fission used twice.
https://www.youtube.com/watch?v=Wkmnwic94Hc - Tool song, Fission used once, SMN AF burn.
https://www.youtube.com/watch?v=s15SPIZiBrU - Fission used once
https://www.youtube.com/watch?v=y7bCxq_KVQ4 - Fission used three times, first usage about 70%
https://www.youtube.com/watch?v=491i_GWHyfM - Fission used once around 1-2%, uploaded 16 years ago
https://www.youtube.com/watch?v=KxYEXR74vH8 - Dragonforce song, Fission used first at 92%, used three times.
https://www.youtube.com/watch?v=Q2rtTKrzqcU - Fission twice
https://www.youtube.com/watch?v=TP4phAP9PP4 - Music, Fission fives times

Promyvion - Vahzl
https://youtu.be/ransAEq98PY?t=322 - Fission once
https://youtu.be/wE8Xqnaed70?t=129 - Fission twice? 
https://youtu.be/VMOhOT8eK8s?t=236 - Fission two times
https://youtu.be/sskqR9RDOng?t=158 - Fission two times
https://youtu.be/MEyaam_40WU?t=168 - Music, Spawned Offspring four times without using Fission*

*The last video is quite interesting.  This party was doing a test run that turned into a clear, and it appears they used almost no animas during the fight.  While watching we can see that it doesn't ever use the TP move Fission according to the player's chatlog.  Other TP moves can readily been seen appearing, so we know this is not an issue of chat filers.  Instead, it seems as though there's a mechanic, possibly caused by Procreator's health being below 20%, that forces it to spawn several Offspring in rapid succession without ever using Fission.

**Issue #3**, this was the reason for creating the original bug report.  It's uncertain exactly what is going wrong in the Promyvion - Vahzl fight against the Procreator, but it does appear that the boss can spawn an add even when it's not supposed to have been able to.  It also seems as though there was some sort of queueing issue going on as well.  This may be the result of two different mechanics being in play.

From the several videos posted, it appears as though Fission can be used at any time by the NM Procreator, but then there's also a second mechanic which is somehow triggered. This second mechanic is entirely separate from the usage of the TP move Fission.  I have an idea as to how this second mechanic functions.  In this [video](https://youtu.be/MEyaam_40WU?t=300) around the 5:06 mark, the final mob Agonizer is brought into play by Procreator's HP falling below 20%.  30 seconds later, several Offspring begin to be spawned by Procreator without Fission ever being used, all in rapid succession.  I don't believe I've ever seen this mechanic on Eden, but I've also only done the Promyvion - Vahzl fight a handful of times.

In short, based on the above, I strongly believe the add spawning mechanics for Procreator and possibly even the same boss in Dem should be audited to ensure they're working properly.

**Comment by DeuceFFXI** [comment]
I would also add another assumption: If Procreator has a time based mechanic once its HP <=20%, then it's likely the other two NMs do as well.

**Comment by DeuceFFXI** [comment]
So I feel this [video](https://www.youtube.com/watch?v=MEyaam_40WU&t=168s) I talked about in length earlier warranted another look.

Firstly, according to the description it was recorded in late 2004, not long after CoP release when this content was first available.  I thought that this being such an early recording of the fight might mean it could differ mechanically from the other videos I posted, if only slightly.  However, a search of the patch notes thru 2008 showed no such changes to this fight ever being made.  I'm not sure exactly what was changed in the BCNMs later in the game's life as I understand there were nerfs post 2010, but I don't know what was done.  So, whether or not the behavior I observed in this video still exists is something that would need to be tested on retail.

Now, as for the behavior itself, the Procreator, animating the Fission tp move without it appearing in logs, was after further review not exactly spammed.  Of course, the first time it was used was roughly 30 seconds after it's HP fell below 20%, the other uses in the following minute didn't have a discernable pattern to them.

For instance the times were as follows:
1. 33 seconds
2. 72 seconds
3. 80 seconds
4. 88 seconds
_*Times are based on the 5:06 mark of the video._

In between these times the Procreator was still using other TP moves.  This [fight](https://youtu.be/a6BDUvNGegs?t=201) uploaded less than two years later if the description of the other video is accurate, shows Procreator first using Fission around 54% and then multiple times in quick succession below 20%.  I believe this aspect of the fight works appropriately on Eden.  That is, below 20% Fission is used far more frequently.

So, whether or not Fission failing to print in chatlogs was an early artifact of the fight is hard to say.   Unfortunately, SE had a habit of making changes without ever noting them in patch notes as many are aware.  However, I'm still unsure about it's rate of using Fission between 20-50% is appropriate, or even if it has access to it as a TP move even at the beginning of the fight, as seen in a few of these videos.  As for it using Fission and seemingly spawning an add while under the effects of a Hysteroanima, I believe that still is worth looking into.

**Comment by 59blargedy** [comment]
for the anima piece, it is not using it while under the effect, it just wasn't under the effect. the duration is currently set to vary between 15 and 30 seconds, so simply wore off.  jp wiki says 40 secs, forum posts and alla say 15-20, bg/ff wikis say 30. to test, would need to get the monster below 25, calculate the number of hits with a specific weapon to get it to tp, do 1 under that hit, use hysteroanima, hit it again. wait.

**Comment by DeuceFFXI** [comment]
Yeah, I definitely agree to better understand how it works would require some testing on retail.  I don't think there were any changes the mechanics of the fights themselves, but I'd have to look through patch notes after 2008 to be sure.

**Comment by 59blargedy** [comment]
Resolved in August 7, 2023 update. from retail testing it has an absurd amt of regain below 25 which is the only thing i can account for the multiple fission usages. There may be more to this but could not replicate era with hours of retail tests.
